src/detect-engine-build.c

```diff
@@ -1425,7 +1425,12 @@ static inline int CreatePortList(DetectEngineCtx *de_ctx, const uint8_t *unique_
                 port = port2 + 1;
             } else if (p1 && p1->single) {
                 SCPortIntervalFindOverlappingRanges(de_ctx, port, port, &it->tree, list);
```

Comment:
Do we still need this `SCPortIntervalFindOverlappingRanges` ?

Reply:
yes. This looks for [port, port]. Next one looks for [port, port2 -1]